### PR TITLE
Enable duecredit for magdir

### LINKED ISSUE #222: Add citation information to DipoleMagDir
The `gravmag.magdir.DipoleMagDir` class is missing citation information. Need to add the reference below to the class and module docstrings:

Oliveira Jr., V. C., D. P. Sales, V. C. F. Barbosa, and L. Uieda (2015), Estimation of the total magnetization direction of approximately spherical bodies, Nonlin. Processes Geophys., 22(2), 215–232, doi:10.5194/npg-22-215-2015.

### PR DESCRIPTION
**Pull request by mtb-za**
Fixes #222 and test for general rollout of #292

Adds the ability to easily get the citation of a function that was used.

To test:
1. Install duecredit: `pip install duecredit`

(This should be doable without installing it globally, but that is not working
for me. I may be missing something here with importing the module, because
this is what `our_duecredit.py` is supposed to enable, if I understand their
docs correctly.)

Then, either:
1. Call the script with `python -m duecredit
gravmag_magdir_dipolemagdir.py`.

OR

1. Set an environment variable: `DUECREDIT_ENABLE=yes`.
2. Run `python cookbook\gravmag_magdir_dipolemagdir.py`.

Any output from the script will be shown first, followed by something like:

    DueCredit Report:
    - Estimates total mag. dir. of approx. spherical bodies /
    fatiando.gravmag.magdir:DipoleMagDir (v 0.4-21-gcb9ce97-dirty) [1]
    - Scientific tools library / numpy (v 1.10.4) [2]

    1 package cited
    0 modules cited
    1 function cited

    References
    ----------

    [1] Oliveira, V.C. et al., 2015. Estimation of the total magnetization direction
    of approximately spherical bodies. Nonlin. Processes Geophys., 22(2),
    pp.215-232.

### Checklist:

- [x] Make tests for new code (at least 80% coverage)
- [x] Create/update docstrings
- [x] Include relevant equations and citations in docstrings
- [x] Docstrings follow the style conventions
- [x] Code follows PEP8 style conventions
- [x] Code and docs have been spellchecked
- [x] Include new dependencies in `doc/install.rst`, `requirements.txt`, `environment.yml`, `ci/requirements-conda.txt` and `ci/requirements-pip.txt`.
- [x] Documentation builds properly (run `cd doc; make` locally)
- [ ] Changelog entry (leave for last to avoid conflicts)

### CODE REVIEW

**Comment by leouieda** [comment]
@mtb-za wow thanks for this! That was very quick.

From what I understood of their docs, you can only get the citation information if you have `duecredit` installed. The stub that we commit to our repository only does something if it can `import duecredit` [[1]](https://github.com/mtb-za/fatiando/blob/ecfb7eff296919fae4d3978c0ee9e9160047e2ac/fatiando/our_duecredit.py#L52). If not, then it does nothing and doesn't bother us. That is a good thing because it doesn't add yet another dependency for us.

**Comment by leouieda** [comment]
While you're at it, could you also please add the following to the docstring of `DipoleMagDir`?

After the first line of the docstring add the line:

    Implements the method of Oliveira Jr. et al. (2015).

and place the reference after Blakely:

    Oliveira Jr., V. C., D. P. Sales, V. C. F. Barbosa, and L. Uieda (2015), 
    Estimation of the total magnetization direction of approximately spherical bodies, 
    Nonlin. Processes Geophys., 22(2), 215–232, doi:10.5194/npg-22-215-2015.

That way we are complete with #222

**Comment by leouieda** [comment]
One more thing, maybe we should rename `our_duecredit.py`  to `_our_duecredit.py`  so that it is hidden from users (this is only meant to be used internally). You can rename in git with `git mv`.

**Comment by mtb-za** [comment]
This should be pretty much done now. If someone gives it the OK, I will edit the changelog and it can be merged.

**Comment by mtb-za** [comment]
I am having trouble running make on this computer. The last commit fixed an error in the Travis build for the docs, so I think it is working now.

I will try and check this in the next couple days on a *nix machine. But if someone else can check before then, that would be good.

The output for the docs from Travis is here:

    building [mo]: targets for 0 po files that are out of date
    building [html]: targets for 122 source files that are out of date
    updating environment: 135 added, 0 changed, 0 removed
    reading sources... [100%] news                                                  
    looking for now-outdated files... none found
    pickling environment... done
    checking consistency... done
    preparing documents... done
    writing output... [100%] news                                                   
    generating indices... genindex py-modindex
    highlighting module code... [100%] fatiando.gravmag.basin2d                     
    writing additional pages... search
    copying images... [100%] gallery/5-seismic/images/thumb/sphx_glr_convolutional_model_thumb.png
    copying downloadable files... [100%] _static/cookbook/mesher_prismmesh.py       
    copying static files... WARNING: favicon file u'favicon.ico' does not exist
    done
    copying extra files... done
    dumping search index in English (code: en) ... done
    dumping object inventory... done
    build succeeded, 1 warning.
    Embedding documentation hyperlinks in examples..
    	processing: index.html
    	processing: point_scatter.html
    	processing: regular_grid.html
    	processing: mayavi-exaggerate.html
    	processing: seismic-wiggle.html
    	processing: prism_mesh.html
    	processing: prism_model.html
    	processing: eqlayer_mag_transform.html
    	processing: eqlayer_transform.html
    	processing: euler_expanding_window.html
    	processing: euler_moving_window.html
    	processing: convolutional_model.html
    	processing: srtomo_regularized.html
    [done]

    Build finished. The HTML pages are in _build/html.
    make: Leaving directory `/home/travis/build/fatiando/fatiando/doc'
    Finished building documentation

File: doc/changelog.rst [hunk 1]
```diff
@@ -40,6 +40,12 @@ Version 0.5
 * Better navigation for long pages in the docs by adding a sidebar with links
   to subsections.
   (`PR 275 <https://github.com/fatiando/fatiando/pull/275>`__)
+* Added back-end support for decorators from duecredit
+  (`<https://github.com/duecredit/duecredit/>`__) to be added to methods. This
```
Comment (leouieda) [inline doc/changelog.rst]: Please replace

    duecredit  (`<https://github.com/duecredit/duecredit/>`__)` 

with 

    `duecredit  <https://github.com/duecredit/duecredit/>`__

The syntax

   `text <target>`__

in rst is equivalent to

    [Text](target)

In Markdown. I get this wrong all the time. That's one of the reasons i added "build the docs locally" to the check list :smile:

**Comment by leouieda** [comment]
Are you getting an error running make or don't have it installed?

Unicode support is one the reasons I really want to move to Python 3. That - in the page numbers of the reference always gets me.

**Comment by mtb-za** [comment]
It is installed, but is giving me an error when I run `make all`:

    sh: C:\Program: No such file or directory
    Makefile:30: recipe for target 'cookbook' failed
    make: *** [cookbook] Error 127

This seems to be something related to a space in the path to `sh`, since the one it is using is `C:\Program Files (x86)\Git\bin\sh.exe`. cf: http://stackoverflow.com/questions/5999507/mingw-make-cant-handle-spaces-in-path
I will run it locally on a *nix machine tomorrow though, which should not have this issue. :)

**Comment by leouieda** [comment]
@mtb-za using command-line tools in Windows is always a pain. OK, I'll run it here as well and see if I catch anything.

**Comment by leouieda** [comment]
@mtb-za Travis seems to be OK and the docs build perfectly on my machine, so I'll merge this now.

Thanks again for doing this so quickly!

**Pull request merged**